addons/project/controllers/portal.py

```diff
@@ -56,12 +56,12 @@ def _project_get_page_view_values(self, project, access_token, page=1, date_begi
         if search and search_in:
             domain += self._task_get_search_domain(search_in, search)
 
-        Task = request.env['project.task']
-        if access_token:
-            Task = Task.sudo()
+        TaskSudo = request.env['project.task'].sudo()
+        if not access_token:
+            domain = AND([domain, request.env['ir.rule']._compute_domain(TaskSudo._name, 'read')])
```

Comment:
After discussing with @IT-Ideas, we should protect the data in the case where this method is used in another way. For instance, another portal route uses this method but not check the user authenticated, so if the current user is a public one then with the sudo this user could see all tasks because the compute domain with the ir.rule will return a empty list since no ir.rule is linked to a public user.

```suggestion
        Task = request.env['project.task']
        if access_token:
            Task = Task.sudo()
        elif not request.env.user.is_public():
            domain = AND([domain, request.env['ir.rule']._compute_domain(Task._name, 'read')])
```

Since the Task will not be always in sudo we cannot named it TaskSudo unfortunately, sorry. :/